src/goose/brokers/redis/api/enqueued_jobs.clj

```diff
@@ -18,15 +18,40 @@
         match? (fn [job] (= (:id job) id))]
     (first (find-by-pattern redis-conn queue match? limit))))
 
-(defn prioritise-execution [redis-conn job]
-  (let [ready-queue (:ready-queue job)]
-    (when (redis-cmds/list-position redis-conn ready-queue job)
-      (redis-cmds/del-from-list-and-enqueue-front redis-conn ready-queue job))))
+(defn prioritise-execution
+  ([redis-conn job]
+   (let [ready-queue (:ready-queue job)]
+     (when (redis-cmds/list-position redis-conn ready-queue job)
+       (redis-cmds/del-from-list-and-enqueue-front redis-conn ready-queue job))))
 
-(defn delete [redis-conn job]
-  (let [queue (job/ready-or-retry-queue job)]
-    (= 1 (redis-cmds/del-from-list redis-conn queue job))))
+  ;; This isn't exposed by Broker API and is used internally by console
+  ([redis-conn queue jobs]
+   (let [remove-jobs-with-invalid-positions-fn (fn [p j] (->> (mapv vector p j)
+                                                              (remove #(nil? (first %)))
+                                                              (mapv second)))
+         positions (redis-cmds/list-position-multiple redis-conn (d/prefix-queue queue) jobs)
+         positions-seq (if (sequential? positions) positions (vector positions))
+         jobs (remove-jobs-with-invalid-positions-fn positions-seq jobs)]
+     (redis-cmds/del-from-list-and-enqueue-front-multiple redis-conn (d/prefix-queue queue) jobs))))
+
+(defn delete
+  ([redis-conn job]
+   (let [queue (job/ready-or-retry-queue job)]
+     (= 1 (redis-cmds/del-from-list redis-conn queue job))))
+
+  ;; Used internally by console
+  ([redis-conn queue jobs]
+   (let [resp (redis-cmds/del-from-list-multiple redis-conn (d/prefix-queue queue) jobs)
+         resp-seq (if (sequential? resp) resp (vector resp))
+         deleted-jobs-count (-> resp-seq
+                                frequencies
+                                (get 1))]
+     (= deleted-jobs-count (count jobs)))))
 
 (defn purge [redis-conn queue]
   (let [ready-queue (d/prefix-queue queue)]
     (= 1 (redis-cmds/del-keys redis-conn ready-queue))))
+
+;;Used internally by console
+(defn get-by-range [redis-conn queue start stop]
```

Comment:
NOTE: This function isn't exposed by the redis API and is used internally used by console.
question: Shall I keep this comment for clarity or it makes it unnecessarily verbose?